input/core/src/commonMain/kotlin/majestic/editor/ToolBarTabs.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.launch
+
+class ToolBarTabColors(
+    val text: StateColors = StateColors(Color.White, Color.Gray),
+    val bottomLine: StateColors = StateColors(Color(0xFF0061FF), Color.Gray)
+)
+
+fun Modifier.borderBottom(
+    color: Color = Color.Black,
+    width: Dp = 2.dp
+) = drawBehind {
+    drawLine(
+        color = color,
+        start = Offset(0f, size.height),
+        end = Offset(size.width, size.height),
+        strokeWidth = width.toPx(),
+    )
+}
+
+@Composable
+fun ToolBarTabs(
+    modifier: Modifier,
+    toolBarHost: Host,
+    colors: ToolBarTabColors = ToolBarTabColors(),
+    fontSize: TextUnit = 13.sp,
+    fontFamily: FontFamily = FontFamily.Default,
+    fontWeight: FontWeight = FontWeight(500)
+) {
+    var selectedTabIndex by remember { mutableStateOf(0) }
+    val coroutineScope = rememberCoroutineScope()
+
+
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.SpaceBetween
+    ) {
+        Row(
+            modifier = Modifier
+                .horizontalScroll(rememberScrollState()),
+            horizontalArrangement = Arrangement.spacedBy(25.dp, Alignment.Start),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            toolBarHost.tabs.forEachIndexed { index, toolBar ->
+                val active = selectedTabIndex == index
+                val activeColor = colors.bottomLine.active
+                val inActiveColor = colors.bottomLine.inActive
+
+                val interactionSource = remember { MutableInteractionSource() }
+                val isHovered by interactionSource.collectIsHoveredAsState()
+                var hovered by remember { mutableStateOf(isHovered) }
+
+                coroutineScope.launch {
+                    interactionSource.interactions.collect {
+                        hovered = it is androidx.compose.foundation.interaction.HoverInteraction.Enter
+                    }
+                }
```

Comment:
Are you sure you need this block of code to determin if something is hovered or not? I think you can just use `isHovered` to get those values. But again, check first and see if that is actually the case